mixer/adapter/stackdriver/contextgraph/workload.go

```diff
@@ -151,56 +168,88 @@ func (wi workloadInstance) Reify(logger adapter.Logger) ([]entity, []edge) {
 	return []entity{wiEntity, owner, workload}, edges
 }
 
+func (s service) Reify() entity {
+	return entity{
+		containerFullName: fmt.Sprintf("//cloudresourcemanager.googleapis.com/projects/%s",
+			s.istioProject),
+		typeName: "io.istio.Service",
+		fullName: fmt.Sprintf("//istio.io/projects/%s/meshes/%s/services/%s/%s",
+			s.istioProject,
+			url.QueryEscape(s.meshUID),
+			url.QueryEscape(s.namespace),
+			url.QueryEscape(s.name)),
+		location: "global",
+		shortNames: [4]string{
+			url.QueryEscape(s.meshUID),
+			url.QueryEscape(s.namespace),
+			url.QueryEscape(s.name),
+			"",
+		},
+	}
+}
+
 type trafficAssertion struct {
 	source, destination          workloadInstance
 	contextProtocol, apiProtocol string
+	destinationService           service
 	timestamp                    time.Time
 }
 
 func (t trafficAssertion) Reify(logger adapter.Logger) ([]entity, []edge) {
-	var sourceFullNames, destinationFullNames []string
+
+	commType, ok := protocolMap[t.contextProtocol]
+	if !ok {
+		if commType, ok = protocolMap[t.apiProtocol]; !ok {
+			logger.Warningf("Unknown type of protocol: %s", t.apiProtocol)
+		}
+	}
+
+	serviceEntity := t.destinationService.Reify()
 	entities, edges := t.source.Reify(logger)
+	var sourceFullNames []string
 	for _, entity := range entities {
 		sourceFullNames = append(sourceFullNames, entity.fullName)
+		if len(commType) > 0 {
```

Comment:
Why to have this if statement? CommType should always be filled when reaching here, right?

Reply:
preserving functionality... this mirrors the `if typeName != "" {` from the existing code. It is possible that `commType` will be nil if the `contextProtocol` and the `apiProtocol` in the `trafficAssertion` are not found in the map (for whatever reason).

Reply:
Sorry, did not realize the code will continue no matter commType is found or not at beginning of function.